src/backend/cuda/subconfigure.m4

```diff
@@ -66,14 +66,15 @@ if test "$with_cuda" != "no" ; then
 EOF
         if test -n "$ac_save_CC" ; then
             NVCC_FLAGS="-ccbin $ac_save_CC"
-            # - pgcc doesn't work, use pgc++ instead
+            # - pgcc/nvc doesn't work, use pgc++/nvc++ instead
             # - Extra optins such as `gcc -std=gnu99` doesn't work, strip the option
             NVCC_FLAGS=$(echo $NVCC_FLAGS | sed -e 's/pgcc/pgc++/g' -e's/ -std=.*//g')
+            NVCC_FLAGS=$(echo $NVCC_FLAGS | sed -e 's/nvc/nvc++/g' -e's/ -std=.*//g')
         else
             NVCC_FLAGS=""
         fi
         # try nvcc from PATH if 'with-cuda' does not contain a valid path
-        if test ${with_cuda} == "yes"; then
+        if test ${with_cuda} = "yes"; then
```

Comment:
Wait, shouldn't this be `!=`?

Reply:
Lol, yeah it should looks like it.

Reply:
Oh wait, I see it now. `yes` isn't a valid path so it just tries to use whatever it can find in `PATH`.

Reply:
Maybe it would be better to do:
```
if test -d ${with_cuda} ; then
    nvcc_bin=${with_cuda}/bin/nvcc
else
    nvcc_bin=nvcc
fi
```

Reply:
Yes, that reads much clearer.

Reply:
Or maybe it is more proper to use `AC_CHECK_PROG`?

Reply:
I went with the `test -d` version. AC_CHECK_PROG seems a little clunky if I am understanding it right.